core/ide/src/main/java/org/alice/ide/frametitle/AliceIdeFrameTitleGenerator.java

```diff
@@ -42,31 +42,32 @@
  *******************************************************************************/
 package org.alice.ide.frametitle;
 
-import edu.cmu.cs.dennisc.java.net.UriUtilities;
 import org.alice.ide.IDE;
+import org.alice.ide.uricontent.UriProjectLoader;
 
 import java.io.File;
-import java.net.URI;
 
 /**
  * @author Dennis Cosgrove
  */
 public class AliceIdeFrameTitleGenerator implements IdeFrameTitleGenerator {
   @Override
-  public String generateTitle(URI uri, boolean isDocumentUpToDateWithUri) {
+  public String generateTitle(UriProjectLoader projectLoader, boolean isDocumentUpToDateWithUri) {
     StringBuilder sb = new StringBuilder();
     sb.append(IDE.getApplicationName());
     sb.append(" ");
     sb.append(IDE.getVersionAdornment());
     sb.append(" ");
-    if (uri != null) {
-      File file = UriUtilities.getFile(uri);
+
+    boolean isBackup = false;
+    if (projectLoader != null) {
+      isBackup = projectLoader.isBackup();
+      File file = projectLoader.getMainProjectFile();
       if (file != null) {
         sb.append(file);
       }
-      sb.append(" ");
     }
-    if (!isDocumentUpToDateWithUri) {
+    if (isBackup || !isDocumentUpToDateWithUri) {
```

Comment:
This change does cause the * to show up, but it removes consistent meaning from it because it does not have any effect as far as closing the world.
If it is a backup and you have not saved it it will not stop you, which is not what the * generally implies.